### PR TITLE
Create class #135: Better integration with main app.

### PR DESCRIPTION
**Pull request by brendanmcgarry**
Schema incompatibility resolved, backend refactored, UI improved and added into main app.

### CODE REVIEW

**Comment by brendanmcgarry** [comment]
The angular for the form is a bit messy/roundabout but it is functional and can be fixed/refactored.

File: app/models/Classes.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ let chatSchema = new Schema({
 
 let classSchema = new Schema({
     _id: String,
+    // courseCode: String,
```
Comment (jusleg) [inline app/models/Classes.js]: we should not keep commented code that is not relevant

File: public/app/app.html [hunk 2]
```diff
@@ -39,6 +39,7 @@
                     </a>
                 </div>
             </div>
+            <a ng-click="vm.openCreateClassModal('md')" ng-if="$root.currentUser.role == 2">Create Class</a>
         </div>
         <hr>
         <!--TODO: do properly with angular routing-->
```
Comment (jusleg) [inline public/app/app.html]: todo should be kept out of the code

File: public/javascript/createclass.controller.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/**
+ * Created by ericxiao on 2017-03-23.
+ */
+/**
+ * Created by ericxiao on 2017-02-09.
```
Comment (jusleg) [inline public/javascript/createclass.controller.js]: Double header?

File: public/javascript/createclass.js [hunk 4]
```diff
@@ -59,33 +59,46 @@ function submitCreateClass() {
...
+	}
 
-	console.log("invalid className? " + isEmpty(className));
-	if(isEmpty(className))
+	if(isEmpty(className)) {
 		validInput = false;
+		errorMsg += "Class Name field must have a value.\n";
+	}
 
 	//if(hours.match(/^(mon|tue|wed|thu|fri|sat|sun)\/$/)) //TODO: hours validation, e.g. tue/17:45/20:15,thurs/8:45/10:00
```
Comment (jusleg) [inline public/javascript/createclass.js]: Are we going to use this? still relevant?

File: public/javascript/createclass.js [hunk 5]
```diff
@@ -59,33 +59,46 @@ function submitCreateClass() {
...
-	console.log("invalid hours? " + isEmpty(hours))
-	if(isEmpty(hours))
+	if(isEmpty(hours)) {
 		validInput = false;
+		errorMsg += "Class Hours field must have a value.\n";
+	}
 
 	//if(location) //TODO: location validation
```
Comment (jusleg) [inline public/javascript/createclass.js]: view todo comment above

File: public/javascript/modules/createclass.module.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Created by ericxiao on 2017-03-23.
+ */
+/**
+ * Created by ericxiao on 2017-01-31.
```
Comment (jusleg) [inline public/javascript/modules/createclass.module.js]: duplicate

**Pull request merged**